data-protocols/dsp/dsp-catalog/dsp-catalog-api/src/main/java/org/eclipse/edc/protocol/dsp/catalog/api/controller/CatalogController.java

```diff
@@ -74,39 +78,46 @@ public CatalogController(Monitor monitor, ObjectMapper mapper, IdentityService i
-        var claimToken = identityService.verifyJwtToken(tokenRepresentation, dspCallbackAddress)
-                .orElseThrow(failure -> new AuthenticationFailedException());
+        try {
+            var tokenRepresentation = TokenRepresentation.Builder.newInstance()
+                    .token(token)
+                    .build();
 
-        var expanded = jsonLdService.expand(jsonObject);
-        if (expanded.failed()) {
-            throw new InvalidRequestException(expanded.getFailureDetail());
-        }
-        var expandedJson = expanded.getContent();
-        if (!isOfExpectedType(expandedJson, DSPACE_CATALOG_REQUEST_TYPE)) {
-            throw new InvalidRequestException(format("Request body was not of expected type: %s", DSPACE_CATALOG_REQUEST_TYPE));
-        }
+            var claimToken = identityService.verifyJwtToken(tokenRepresentation, dspCallbackAddress)
+                    .orElseThrow(failure -> new AuthenticationFailedException());
+
+            var expanded = jsonLdService.expand(jsonObject);
+            if (expanded.failed()) {
+                throw new InvalidRequestException(expanded.getFailureDetail());
+            }
+            var expandedJson = expanded.getContent();
+            if (!isOfExpectedType(expandedJson, DSPACE_CATALOG_REQUEST_TYPE)) {
+                throw new InvalidRequestException(format("Request body was not of expected type: %s", DSPACE_CATALOG_REQUEST_TYPE));
+            }
 
-        var message = transformerRegistry.transform(expandedJson, CatalogRequestMessage.class)
-                .orElseThrow(failure -> new InvalidRequestException(format("Request body was malformed: %s", failure.getFailureDetail())));
+            var message = transformerRegistry.transform(expandedJson, CatalogRequestMessage.class)
+                    .orElseThrow(failure -> new InvalidRequestException(format("Request body was malformed: %s", failure.getFailureDetail())));
 
-        var catalog = service.getCatalog(message, claimToken)
-                .orElseThrow(exceptionMapper(Catalog.class));
+            var catalog = service.getCatalog(message, claimToken)
+                    .orElseThrow(exceptionMapper(Catalog.class));
 
-        var catalogJson = transformerRegistry.transform(catalog, JsonObject.class)
-                .orElseThrow(failure -> new EdcException(format("Failed to build response: %s", failure.getFailureDetail())));
+            var catalogJson = transformerRegistry.transform(catalog, JsonObject.class)
+                    .orElseThrow(failure -> new EdcException(format("Failed to build response: %s", failure.getFailureDetail())));
 
-        var compacted = jsonLdService.compact(catalogJson);
-        if (compacted.failed()) {
-            throw new InvalidRequestException(compacted.getFailureDetail());
+            var compacted = jsonLdService.compact(catalogJson);
+            if (compacted.failed()) {
+                throw new InvalidRequestException(compacted.getFailureDetail());
```

Comment:
if it fails in compaction is not a request exception, better go with an EdcException (500)